### PR TITLE
Fix "Mark all as read" button for local notifications

### LINKED ISSUE #291: "Mark all as read" button should clear extension's unread notifications
"Mark all as read" button does not clear notifications, that were "remembered" by the extension's "Mark as unread" feature.

screen shot 2016-09-04 at 7 22 57 pm

See https://github.com/sindresorhus/refined-github/pull/284 for "mark as unread" feature.

### PR DESCRIPTION
**Pull request by AshikNesin**
### Issue
"Mark all as read" button does not clear notifications that were added using extention

### Fix
Clear localStorage.unreadNotifications when button is clicked.

### Screenshot of Fix
screen recording 2017-05-14 at 09 05 pm

Fixes #291

### CODE REVIEW

File: extension/mark-unread.js [hunk 1]
```diff
@@ -254,6 +254,9 @@ window.markUnread = (() => {
 			renderNotifications();
 			$(document).on('click', '.js-mark-read', markNotificationRead);
 			$(document).on('click', '.js-mark-all-read', markAllNotificationsRead);
+			$(document).on('click', 'form[action="/notifications/mark"] button', () => {
+				localStorage.unreadNotifications = [];
```
Comment (jgierer12) [inline extension/mark-unread.js]:
```js
localStorage.unreadNotifications = JSON.stringify([]);
```
or
```js
localStorage.unreadNotifications = '[]';
```

Comment (AshikNesin) [inline extension/mark-unread.js]: oops... Totally forgot it 😅

Fixing it now.

**Comment by jgierer12** [comment]
Also, the button isn't even shown if there are only local notifications. Can you work on that too @AshikNesin? It should be just a matter of manually inserting the button and form HTML if it doesn't exist. You can do it in this PR since it's closely related

**Comment by AshikNesin** [comment]
@jgierer12 Yeah.. That makes sense. Will do it :)

**Comment by AshikNesin** [comment]
@jgierer12 

I just noticed that we have similar feature. (After clicking it cleans local notifications)


Do we need another button for the similar button?

(Correct me if I'm wrong 😄)

**Comment by jgierer12** [comment]
The checkmark button is only for a specific repo. The "mark all as read" button at the top clears all notifications from all repos.

**Comment by AshikNesin** [comment]
@jgierer12 My bad didn't think about it. Will do that tomorrow :)

**Comment by AshikNesin** [comment]
@jgierer12 

Added custom "mark all read" button" :)

**Comment by paulmolluzzo** [comment]
@jgierer12 Do you approve this PR? If so you can squash and merge.

:shipit:

**Comment by jgierer12** [comment]
Great work @AshikNesin 👏

**Pull request merged**